beacon-chain/state/fieldtrie/field_trie_helpers.go

```diff
@@ -141,3 +150,47 @@ func handlePendingAttestation(val []*ethpb.PendingAttestation, indices []uint64,
 	}
 	return roots, nil
 }
+
+func handleBalanceSlice(val []uint64, indices []uint64, convertAll bool) ([][32]byte, error) {
```

Comment:
I agree with Radek's comment that the control variable is a little cryptic, especially since it's actually passed through indirectly (going through `fieldConverters` first). Maybe it would be more clear if the bool indicated *why* you wanted to convert all, like if it was named 'recompute' (where `!recompute` == `convertAll`).

Reply:
Actually looking at these methods more closely, I think they all early return in the `convertAll` path, so it does seem like it would make more sense for them to be separate functions.

Reply:
I can handle this in a follow up PR, since all our other methods also follow the same control flow.